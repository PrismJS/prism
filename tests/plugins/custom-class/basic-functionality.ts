import { createTestSuite } from '../../helper/prism-dom-util';


describe('Custom class', () => {
	const { it } = createTestSuite({
		languages: 'javascript',
		plugins: 'custom-class'
	});

	it('should set prefix', ({ Prism, util }) => {
		Prism.plugins.customClass.prefix = 'prism-';

		util.assert.highlight({
			language: 'javascript',
			code: `var a = true;`,
		});
	});

	it('should reset prefix', ({ Prism, util }) => {
		Prism.plugins.customClass.prefix = '';

		util.assert.highlight({
			language: 'javascript',
			code: `var a = true;`,
		});
	});

	it('should map class names using a function', ({ Prism, util }) => {
		Prism.plugins.customClass.map((cls) => {
			return `${cls}-suffix`;
		});

		util.assert.highlight({
			language: 'javascript',
			code: `var a = true;`,
		});
	});

	it('should map class names using an object', ({ Prism, util }) => {
		Prism.plugins.customClass.map({
			boolean: 'b',
			keyword: 'kw',
			operator: 'op',
			punctuation: 'p'
		});

		util.assert.highlight({
			language: 'javascript',
			code: `var a = true;`,
		});
	});

	it('should reset map', ({ Prism, util }) => {
		Prism.plugins.customClass.map({});

		util.assert.highlight({
			language: 'javascript',
			code: `var a = true;`,
		});
	});

});
