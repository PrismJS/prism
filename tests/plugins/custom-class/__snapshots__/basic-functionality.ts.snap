// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom class should map class names using a function 1`] = `
"<span class=\\"token-suffix keyword-suffix\\">var</span>
a
<span class=\\"token-suffix operator-suffix\\">=</span>
<span class=\\"token-suffix boolean-suffix\\">true</span>
<span class=\\"token-suffix punctuation-suffix\\">;</span>
"
`;

exports[`Custom class should map class names using an object 1`] = `
"<span class=\\"token kw\\">var</span>
a
<span class=\\"token op\\">=</span>
<span class=\\"token b\\">true</span>
<span class=\\"token p\\">;</span>
"
`;

exports[`Custom class should reset map 1`] = `
"<span class=\\"token keyword\\">var</span>
a
<span class=\\"token operator\\">=</span>
<span class=\\"token boolean\\">true</span>
<span class=\\"token punctuation\\">;</span>
"
`;

exports[`Custom class should reset prefix 1`] = `
"<span class=\\"token keyword\\">var</span>
a
<span class=\\"token operator\\">=</span>
<span class=\\"token boolean\\">true</span>
<span class=\\"token punctuation\\">;</span>
"
`;

exports[`Custom class should set prefix 1`] = `
"<span class=\\"prism-token prism-keyword\\">var</span>
a
<span class=\\"prism-token prism-operator\\">=</span>
<span class=\\"prism-token prism-boolean\\">true</span>
<span class=\\"prism-token prism-punctuation\\">;</span>
"
`;
